### PR TITLE
Let cabanacore and Cajita be header-only libraries

### PR DESCRIPTION
**Pull request by dalg24**
Partially addresses #361

### CODE REVIEW

File: core/src/Cabana_Version.hpp [hunk 1]
```diff
@@ -19,9 +19,9 @@
 namespace Cabana
 {
 
-std::string version();
+std::string version() { return Cabana_VERSION_STRING; }
```
Comment (streeve) [inline core/src/Cabana_Version.hpp:22]: These need to be inlined to build downstream

Comment (dalg24) [reply core/src/Cabana_Version.hpp:22]: Good catch.  Multiple definition error at link time otherwise

**Review by streeve — APPROVED** [review]
Working with CabanaMD

**Comment by dalg24** [comment]
> Working with CabanaMD

Thank you for checking

**Review by junghans — APPROVED** [review]

**Comment by junghans** [comment]
Yeah to modern CMake!

**Review by sslattery — APPROVED** [review]

**Pull request merged**